### ISSUE TITLE
Not all gpu have nvlink, the communication data is all incorrect

### ISSUE THREAD

**Issue by zhaowujin** [NVIDIA/nccl#1423]
If I have an eight gpus machine, only the 6 and 7 gpu have nvlinks between them, and the other cards do not. Only 6 and 7 can communicate normally, and the communication data between other gpus is all incorrect. I think in this situation, automatic recognition and PCIE should be used.
The detailed errors are as follows:

![Image](https://github.com/user-attachments/assets/22595f25-e4e5-42bb-8d44-2a4a0821a4f2)

6 and 7 GPU is normal.

![Image](https://github.com/user-attachments/assets/e9c1c299-e64f-4f8b-bbf1-ec98ee20b59e)

Other gpus is all incorrect

**Comment by sjeaugey** [comment]
Could it be an ACS issue?

https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/troubleshooting.html#pci-access-control-services-acs

**Comment by zhaowujin** [comment]
> Could it be an ACS issue?
> 
> https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/troubleshooting.html#pci-access-control-services-acs

That means I can only choose to connect the 8 GPUs through nvlink or have all 8 GPUs connected through PCIe？

**Comment by sjeaugey** [comment]
No, that was just asking to check whether ACS was enabled (if running baremetal) and if it was enabled, try to disable it and see if it fixes the issue.

**Comment by zhaowujin** [comment]
> No, that was just asking to check whether ACS was enabled (if running baremetal) and if it was enabled, try to disable it and see if it fixes the issue.

Finally, I closed IOMMU in the BIOS. For specific operation reference, https://github.com/pytorch/pytorch/issues/84803 . thank you!
